app/containers/write.js

```diff
@@ -100,6 +106,40 @@ class writeScreen extends Component {
     }
   }
 
+  viewNextScreen () {
+    const position = this.props.screenViewHistoryPosition + 1
+    const nextScreen = this.props.screenViewHistory[position]
+    if (nextScreen) {
+      if (this.props.addr === nextScreen.addr) {
+        this.props.viewChannel({ addr: nextScreen.addr, channel: nextScreen.channel, skipScreenHistory: true })
+      } else {
+        this.props.viewCabal({ addr: nextScreen.addr, channel: nextScreen.channel, skipScreenHistory: true })
+      }
+      this.props.setScreenViewHistoryPostion({ index: position })
+    } else {
+      this.props.setScreenViewHistoryPostion({
+        index: this.props.screenViewHistory.length - 1
+      })
+    }
+  }
+
+  viewPreviousScreen () {
+    const position = this.props.screenViewHistoryPosition - 1
+    const previousScreen = this.props.screenViewHistory[position]
+    if (previousScreen) {
+      if (this.props.addr === previousScreen.addr) {
+        this.props.viewChannel({ addr: previousScreen.addr, channel: previousScreen.channel, skipScreenHistory: true })
+      } else {
+        this.props.viewCabal({ addr: previousScreen.addr, channel: previousScreen.channel, skipScreenHistory: true })
+      }
+      this.props.setScreenViewHistoryPostion({ index: position })
+    } else {
+      this.props.setScreenViewHistoryPostion({
+        index: 0
+      })
+    }
+  }
+
```

Comment:
@nikolaiwarner a maybe cleaner way to do it is to keep track if we're typing instead? and inside `viewPreviousScreen ()`, `viewNextScreen ()` have something like
```js
// don't change screens while the typing area is focused or has been typed in recently 
if (this.props.typing) { return }
```

i guess i was just a bit confused around the var `skipScreenHistory` and what it did

just a tiny bit of code review! :3 feel free to merge it if that makes more sense in yr opionion 🧅

Reply:
also i recently woke up so i should probably be drinking tea in the morning sun to wake up instead of like review code lol

Reply:
🍵 

The `skipScreenHistory` is to tell the viewCabal and viewChannel functions that you are walking through history and not to add that navigation event to the end of the history stack so that navigating forward through history works.

Reply:
ahhhhhh i _completely_ misunderstood, sorriii 🙏 

anyway might be worth to leave a shortened version of yr reply here as a comment in `viewCabal` and `viewChannel` for future contribs ^_^